nbodykit/base/catalog.py

```diff
@@ -1165,6 +1165,117 @@ def csize(self):
             return NotImplemented
         return self._csize
 
+    def gslice(self, start, stop, end=1):
+        """
+        Execute a global slice of a CatalogSource.
+
+        .. note::
+            After the global slice is performed, the data is scattered
```

Comment:
I wonder if it is useful to avoid the scattering with an option?

Reply:
I am not sure I can think of a use case now, but it doesn't hurt to include the option, I think

Reply:
Avoiding the scattering probably means the compute can be deferred further?

Reply:
yes I think so. I wonder if it is possible to implement this with no compute() calls?  We can't use Scatter/GatherArray for dask arrays but we can bcast the appropriate slices of them?